src/main/java/com/tterrag/k9/commands/CommandCurseGradle.java

```diff
@@ -0,0 +1,346 @@
+package com.tterrag.k9.commands;
+
+import java.awt.Color;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.SocketTimeoutException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Random;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import com.tterrag.k9.commands.api.Argument;
+import com.tterrag.k9.commands.api.Command;
+import com.tterrag.k9.commands.api.CommandBase;
+import com.tterrag.k9.commands.api.CommandContext;
+import com.tterrag.k9.commands.api.CommandException;
+import com.tterrag.k9.commands.api.Flag;
+import com.tterrag.k9.util.Threads;
+
+import lombok.extern.slf4j.Slf4j;
+import sx.blah.discord.handle.obj.IMessage;
+import sx.blah.discord.util.EmbedBuilder;
+import sx.blah.discord.util.RateLimitException;
+import sx.blah.discord.util.RequestBuffer;
+
+/**
+ * Command to figure out curseforges gradle and dependencies.
+ * @author Wyn Price
+ *
+ */
+@Command
+@Slf4j
+public class CommandCurseGradle extends CommandBase {
+    
+    private class CurseResult 
+    {
+        private final String URL;
+        private final String gradle;
+        
+        public CurseResult(String URL, String gradle) 
+        {
+            this.URL = URL;
+            this.gradle = gradle;
+        }
+        
+        public String getGradle() {
+            return gradle;
+        }
+        
+        public String getURL() {
+            return URL;
+        }
+        
+        @Override
+        public String toString() {
+            return "URL:" + URL + "\nGRADLE: " + gradle + "\n";
+        }
+    }
+    
+    private static final Argument<String> ARG_FILEURL = new WordArgument("fileurl", "The URL of the CurseForge project to generate a dependency list for.", true);
```

Comment:
Should probably have a new regex for URLs, just something simple like `https?://\S+\.\w+`

Reply:
would I have to set up my own `SimpleArgument` class or do you have one that supports regex?